From 098bee1120dd5ee13830a04999901c6b3d415541 Mon Sep 17 00:00:00 2001
From: Allan Reyes <allanbreyes@users.noreply.github.com>
Date: Fri, 18 May 2018 00:33:30 -0700
Subject: [PATCH] Safely parse response.unread; fixes #691

---
 plugins/backend/ttrss/ttrssAPI.vala | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/plugins/backend/ttrss/ttrssAPI.vala b/plugins/backend/ttrss/ttrssAPI.vala
index 582687f9..1a99d934 100644
--- a/plugins/backend/ttrss/ttrssAPI.vala
+++ b/plugins/backend/ttrss/ttrssAPI.vala
@@ -191,7 +191,15 @@ public class FeedReader.ttrssAPI : GLib.Object {
 		if(error == ConnectionError.SUCCESS)
 		{
 			var response = message.get_response_object();
-			unread = int.parse(response.get_string_member("unread"));
+			int64 parsed = 0;
+			bool res = int64.try_parse(response.get_string_member("unread"), out parsed);
+			if(res) {
+				unread = (int) parsed;
+			}
+			else
+			{
+				Logger.warning("Could not parse unread articles");
+			}
 		}
 		Logger.info("There are %i unread articles".printf(unread));
 
